siliconcompiler/fpgas/vpr_example.py

```diff
@@ -69,6 +70,30 @@ def setup(chip):
             fpga.add('fpga', part_name, 'file', 'yosys_flop_techmap',
                      os.path.join(techlib_root, 'example_arch_techmap_flops.v'))
 
+            fpga.add('fpga', part_name, 'file', 'yosys_dsp_techmap',
+                     os.path.join(techlib_root, 'example_arch_techmap_dsp.v'))
+
+            fpga.add('fpga', part_name, 'file', 'yosys_dsp_extractlib',
+                     os.path.join(techlib_root, 'example_arch_techmap_dsp_extract.v'))
+
+            # This techlib combines the blackboxing and extraction in one library
+            # with a +define; build it out accordingly
+            fpga.add('fpga', part_name, 'file', 'yosys_dsp_blackboxlib',
+                     os.path.join(techlib_root, 'example_arch_techmap_dsp_extract.v'))
```

Comment:
why are two needed?

Reply:
In the new implementation, this will be needed to trigger the file to be read both at the read_verilog step and the extract step; it's not difficult to imagine a setup where you would want to use a given file for only one and not the other, but in this example one file is suitable for both.